is31fl3731/is31fl3731.go

```diff
@@ -0,0 +1,303 @@
+				return err
+			}
+		} else {
+			// Enable every LED (16 columns x 9 rows)
+			for i := uint8(0); i < 16; i++ {
+				err = d.bus.WriteRegister(d.Address, i, []byte{0xFF})
+				if err != nil {
+					return err
+				}
+			}
+		}
+	}
+
+	return nil
+}
+
+// setPixelPWD sets individual pixel's PWM value [0-255] on the selected frame
+func (d *Device) setPixelPWD(frame, n, value uint8) (err error) {
+	err = d.selectCommand(frame)
+	if err != nil {
+		return err
+	}
+
+	return d.bus.WriteRegister(d.Address, LED_PWM_OFFSET+n, []byte{value})
+}
+
+// SetActiveFrame sets frame to display with LEDs
+func (d *Device) SetActiveFrame(frame uint8) (err error) {
+	if frame > FRAME_7 {
+		return fmt.Errorf("frame %d is out of valid range [0-7]", frame)
+	}
+
+	return d.writeFunctionRegister(SET_ACTIVE_FRAME, []byte{frame})
+}
+
+// Fill the whole frame with provided PWM value [0-255]
+func (d *Device) Fill(frame, value uint8) (err error) {
+	if frame > FRAME_7 {
+		return fmt.Errorf("frame %d is out of valid range [0-7]", frame)
+	}
+
+	err = d.selectCommand(frame)
+	if err != nil {
+		return err
+	}
+
+	data := make([]byte, 24)
+	for i := range data {
+		data[i] = value
+	}
+
+	for i := uint8(0); i < 6; i++ {
+		err = d.bus.WriteRegister(d.Address, LED_PWM_OFFSET+i*24, data)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// Clear the whole frame
+func (d *Device) Clear(frame uint8) (err error) {
+	return d.Fill(frame, 0x00)
+}
+
+// DrawPixelIndex draws a single pixel on the selected frame by its index with
+// provided PWM value [0-255]
+func (d *Device) DrawPixelIndex(frame, index, value uint8) (err error) {
+	if frame > FRAME_7 {
+		return fmt.Errorf("frame %d is out of valid range [0-7]", frame)
+	}
+
+	return d.setPixelPWD(frame, index, value)
+}
+
+// DrawPixelXY draws a single pixel on the selected frame by its XY coordinates
+// with provided PWM value [0-255]
+func (d *Device) DrawPixelXY(frame, x, y, value uint8) (err error) {
+	var index uint8
+
+	if d.board == boardAdafruitCharlieWing15x7 {
+		if x >= 15 {
+			return fmt.Errorf("invalid value: X is out of range [0, 15]")
+		} else if y >= 7 {
+			return fmt.Errorf("invalid value: Y is out of range [0, 7]")
+		}
+
+		if x < 8 {
+			index = 16*x + y + 1
+		} else { //          ^-- board is one pixel shorter (7 vs 8 pixels)
+			index = 16*(16-x) - y - 1 - 1
+		} //                      ^-- board is one pixel shorter (7 vs 8 pixels)
+	} else {
+		index = 16*x + y
+	}
+
+	return d.setPixelPWD(frame, index, value)
+}
+
+// New creates a raw driver w/o any preset board layout
+func New(bus drivers.I2C, address uint8) (d Device, err error) {
+	d = Device{
+		Address: address,
+		bus:     bus,
+		board:   boardRaw,
+	}
+
+	err = d.configure()
+
+	return d, err
+}
+
+// NewAdafruitCharlieWing15x7 creates a new driver with Adafruit 15x7
+// CharliePlex LED Matrix FeatherWing (CharlieWing) layout
+func NewAdafruitCharlieWing15x7(bus drivers.I2C, address uint8) (d Device, err error) {
```

Comment:
This seem to be very specific, I'd extract all code related to CharlieWing to its own file, like `is31fl3731_acw15x7.go`

Reply:
Moved all CharlieWing code to a separate file (struct) 👍